cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -517,46 +521,46 @@ private boolean deleteSpecificToolVersionCache(String org, String name, String v
         return ProjectUtils.deleteDirectory(toolPath);
     }
 
-//    /**
-//     * Search for tools in central.
-//     *
-//     * @param keyword search keyword.
-//     */
-//    private void searchToolsInCentral(String keyword) {
-//        try {
-//            Settings settings;
-//            try {
-//                settings = RepoUtils.readSettings();
-//                // Ignore Settings.toml diagnostics in the search command
-//            } catch (SettingsTomlException e) {
-//                // Ignore 'Settings.toml' parsing errors and return empty Settings object
-//                settings = Settings.from();
-//            }
-//            CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
-//                    initializeProxy(settings.getProxy()),
-//                    getAccessTokenOfCLI(settings));
-//            ToolSearchResult toolSearchResult = client.searchTool(keyword,
-//                    JvmTarget.JAVA_11.code(),
-//                    RepoUtils.getBallerinaVersion());
-//
-//            List<Tool> tools = toolSearchResult.getTools();
-//            if (tools != null && tools.size() > 0) {
-//                printTools(toolSearchResult.getTools(), RepoUtils.getTerminalWidth());
-//            } else {
-//                outStream.println("no tools found.");
-//            }
-//        } catch (CentralClientException e) {
-//            String errorMessage = e.getMessage();
-//            if (null != errorMessage && !"".equals(errorMessage.trim())) {
-//                // removing the error stack
-//                if (errorMessage.contains("\n\tat")) {
-//                    errorMessage = errorMessage.substring(0, errorMessage.indexOf("\n\tat"));
-//                }
-//                CommandUtil.printError(this.errStream, errorMessage, null, false);
-//                CommandUtil.exitError(this.exitWhenFinish);
-//            }
-//        }
-//    }
+    /**
+     * Search for tools in central.
+     *
+     * @param keyword search keyword.
+     */
+    private void searchToolsInCentral(String keyword) {
+        try {
+            Settings settings;
+            try {
+                settings = RepoUtils.readSettings();
+                // Ignore Settings.toml diagnostics in the search command
+            } catch (SettingsTomlException e) {
+                // Ignore 'Settings.toml' parsing errors and return empty Settings object
+                settings = Settings.from();
+            }
+            CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
+                    initializeProxy(settings.getProxy()),
```

Comment:
@Thevakumar-Luheerathan is there anything more for the proxy implementation?

Reply:
@Thevakumar-Luheerathan please check if it's correct now.

Reply:
ya... It's